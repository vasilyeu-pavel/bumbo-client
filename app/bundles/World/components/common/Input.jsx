import React, { PropTypes } from 'react';

const Input = props => (
  <div className='order-form-control'>
    <label className='order-form-control__label'>{props.label}</label>
    <input {...props} className='order-form-control__input' />
  </div>
);

Input.propTypes = {
  label: PropTypes.string,
};

Input.defaultProps = {
  type: 'text',
};

export default Input;
