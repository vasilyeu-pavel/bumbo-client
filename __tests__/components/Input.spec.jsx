/* global expect, jest */

import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Input from '../../app/bundles/World/components/common/Input';

describe('<Input /> component', () => {
  const minProps = { label: 'name' };

  it('should render input element with label', () => {
    const input = renderer.create(<Input {...minProps} />).toJSON();

    expect(input).toMatchSnapshot();
  });

  it('should render input with required (or any) attribute', () => {
    const input = renderer.create(<Input {...minProps} required='required' />).toJSON();

    expect(input).toMatchSnapshot();
  });

  it('should render input type "email"', () => {
    const input = renderer.create(<Input {...minProps} type='email' />).toJSON();

    expect(input).toMatchSnapshot();
  });

  it('should call the "onChange" method after input change', () => {
    const handler = jest.fn();
    const input = mount(<Input {...minProps} onChange={handler}/>);

    input.find('input').simulate('change');

    expect(handler).toHaveBeenCalled();
  });
});
