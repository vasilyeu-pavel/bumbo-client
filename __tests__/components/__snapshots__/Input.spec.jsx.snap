exports[`<Input /> component should render input element with label 1`] = `
<div
  className="order-form-control">
  <label
    className="order-form-control__label">
    name
  </label>
  <input
    className="order-form-control__input"
    label="name"
    type="text" />
</div>
`;

exports[`<Input /> component should render input type "email" 1`] = `
<div
  className="order-form-control">
  <label
    className="order-form-control__label">
    name
  </label>
  <input
    className="order-form-control__input"
    label="name"
    type="email" />
</div>
`;

exports[`<Input /> component should render input with required (or any) attribute 1`] = `
<div
  className="order-form-control">
  <label
    className="order-form-control__label">
    name
  </label>
  <input
    className="order-form-control__input"
    label="name"
    required="required"
    type="text" />
</div>
`;
